### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by Lichapa**
**In this milestone, I have done the following:**

- Prompt players to enter names and store the names in Variable

- Assign a symbol to players

- Tell the player whose turn it is.

- Show the step a player will need to play the game

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,102 @@
...
+class Game
+  # Start a new game by geting player names
+  def initialize
+    @player = Player.new
+    @board = Board.new
+  end
+
+  def play
+    moves = 0
+    while moves < 9
```
Comment (krys2fa) [inline bin/main.rb]: * [ ] [REQUIRED]  Using out-of-nowhere numbers to terminate a loop (for example, `5.times do ... end`) is a signal that something is wrong. Please use a terminating variable instead. See below for an example where `game_on` is the terminating variable:
```
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
```

**Review by krys2fa — CHANGES_REQUESTED** [review]
## Nicely done @Lichapa 👏 
You did really great in this milestone. Your PR title and description is well-written as well. 💪 
You have a professional-looking README in place even though it is not required in this milestone. 👍 

Just a few more changes required to complete the milestone. They can be seen below:
### Status: Changes Required ♻️ 

- [ ] [OPTIONAL] The game accepts empty spaces as a player name.

Please request for another review after implementing the changes.

Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,103 @@
...
+      puts 'X, Enter 1-9 to play:'
+      player_x_move = gets.chomp
+      puts 'O, Enter 1-9 to play:'
+      player_o_move = gets.chomp
+
+      puts "x played #{player_x_move} And o played #{player_o_move}"
+
+      game_over = true if winner
+    end
+  end
```
Comment (blackpintz) [inline bin/main.rb:74]: You have mapped out scenarios of how the game tells which player turn it is. Also within this loop, map out the following scenarios:

- what does it look like when a player inputs an invalid move? Also, using puts/ print, indicate the message the player gets for an invalid move.

- What does it look like when the game is drawn? Also, using puts/prints, indicate the message the players receive when the game ends in a draw.

- What does it look like when the game is won? Also,  using puts/prints, indicate the message the player receives when they win a game.

**Review by ghost — CHANGES_REQUESTED** [review]
###

File: bin/main.rb [hunk 3]
```diff
@@ -1,2 +1,103 @@
...
+
+  # Display board after a player has played
+  def display_board(board)
+    puts "#{board[0]} | #{board[1]} | #{board[2]}"
+    puts '---------'
+    puts "#{board[3]} | #{board[4]} | #{board[5]}"
+    puts '---------'
+    puts "#{board[6]} | #{board[7]} | #{board[8]}"
+  end
+end
```
Comment (ghost) [inline bin/main.rb]: [Optional] You can add this inside the game loop you created in `play` method.

File: bin/main.rb (hunk 2 above)
Comment (ghost) [inline bin/main.rb]: ## Mandatory

You have mapped out scenarios of how the game tells which player turn it is. Also within this loop, map out the following scenarios:

- what does it look like when a player inputs an invalid move? Also, using puts/ print, indicate the message the player gets for an invalid move.

- What does it look like when the game is drawn? Also, using puts/prints, indicate the message the players receive when the game ends in a draw.

- What does it look like when the game is won? Also,  using puts/prints, indicate the message the player receives when they win a game.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🤓

## Hello team 😃,

**Great job implementing the requested changes 💪. However, there are a few fixes needed as described in the comment below.**

## Happy coding 🥂 .

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED ♻️ 🛑 
HELLO, @Lichapa  👋🏿  GREAT PROGRESS. I LOVE THAT YOU ADDED A README TO YOUR PROJECT. 👏🏿 HOWEVER, THERE ARE STILL SOME CHANGES THAT NEEDS TO BE DONE. KINDLY CHECK BELOW ⬇️ 

- [ ] There are some rubocop offenses, kindly fix all the rubocop offenses ⬇️ 
Screenshot from 2020-12-11 08-31-10

- [ ] [REQUIRED] game tells which player turn it is (without computing who the winner is). I can see you have a method for this in your project, however, you should try and mock it a bit more. what I mean by this is you don't have to write the logic but at least your method should print something.
**hint:** Try and utilize your loop. Remember no logic is needed.

- [ ] `[REQUIRED]` game asks the player to select from available moves. Just like I have stated I can see you have a method for this, however, try and mock it a bit more, remember no logic is required but at least let your method output something. 

- [ ] `[REQUIRED]` game displays board after the player move (without computing which spots on the board are taken). After every player moves, display the board.
**hint:** you can do this by calling your board method inside your loop.

- [ ] `[REQUIRED]` game informs the player if the selected move is a winning move (without computing the winning move). I can see you have the method for this, again you don't need the logic but mock your method a bit more.

**hint:** Utilize the power of puts and print.

### if you have any question, kindly reach me on slack @peterrobert

### HAPPY CODING 💻

**Review by Srjnnnn — APPROVED** [review]
## Status Approved :partying_face: :tada: 
Hello @Lichapa. Congratulations on making the requested changes. I've checked your game and satisfies the conditions that I write down below;
* [ ] game asks players for their names
* [ ] `[REQUIRED]` game tells which player turn it is 
* [ ] `[REQUIRED]` game asks the player to select from available moves
* [ ] `[REQUIRED]` game informs player if the selected move is invalid 
* [ ] `[REQUIRED]` game displays board after the player move 
* [ ] `[REQUIRED]` game informs player if the selected move is a winning move 
* [ ] `[REQUIRED]` game informs player if the selected move is a draw move 
* [ ] `[REQUIRED]` game repeats all actions for the next player's move

### Optional

- [ ]  You can write separate files for the classes that you use. In the next milestones, it'll be asked to do it like that.
- [ ] Instead of commits like add methods you can be more descriptive on your commit messages in the future.

Congratulations again. :+1: You can merge this PR and move on to the next one.
_Eyüp Sercan UYGUR_

**Pull request merged**